core/amber/src/main/python/core/architecture/packaging/output_manager.py

```diff
@@ -73,6 +94,17 @@ def get_port(self, port_id=None) -> WorkerPort:
     def get_output_channel_ids(self):
         return self._channels.keys()
 
+    def save_tuple_to_storage_if_needed(self, amber_tuple: Tuple, port_id=None) -> None:
+        if port_id is None:
+            for writer_thread in self._port_result_writers.values():
```

Comment:
please leave comments to explain the expected behavior of having port id vs not having port id.

Reply:
Added documentation for this method.